src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -135,33 +136,36 @@ export function uiRenderMixin(kbnServer, server, config) {
                   .reverse(),
               ]),
         ];
+        const uiPluginIds = [...kbnServer.newPlatform.__internals.uiPlugins.public.keys()];
 
         const jsDependencyPaths = [
           ...UiSharedDeps.jsDepFilenames.map(
-            filename => `${regularBundlePath}/kbn-ui-shared-deps/${filename}`
+            filename => `${regularBundlePath}/${buildHash}/kbn-ui-shared-deps/${filename}`
           ),
-          `${regularBundlePath}/kbn-ui-shared-deps/${UiSharedDeps.jsFilename}`,
+          `${regularBundlePath}/${buildHash}/kbn-ui-shared-deps/${UiSharedDeps.jsFilename}`,
           ...(isCore
             ? []
             : [
                 `${dllBundlePath}/vendors_runtime.bundle.dll.js`,
                 ...dllJsChunks,
                 `${regularBundlePath}/commons.bundle.js`,
               ]),
-          `${regularBundlePath}/plugin/kibanaUtils/kibanaUtils.plugin.js`,
-          `${regularBundlePath}/plugin/esUiShared/esUiShared.plugin.js`,
-          `${regularBundlePath}/plugin/kibanaReact/kibanaReact.plugin.js`,
-        ];
 
-        const uiPluginIds = [...kbnServer.newPlatform.__internals.uiPlugins.public.keys()];
+          ...['kibanaUtils', 'esUiShared', 'kibanaReact', ...uiPluginIds].map(
+            pluginId => `${regularBundlePath}/${buildHash}/plugin/${pluginId}/${pluginId}.plugin.js`
```

Comment:
Any reason to use the hash key in the path instead of just in the query string? I think

```ts
pluginId => `${regularBundlePath}/plugin/${pluginId}/${pluginId}.plugin.js?_v=${buildHash}`
```

would perform the same invalidation mechanism without having to tweak `buildRouteForBundles` in the optimizer?

Reply:
I remember some problems with proxy server were reported
https://css-tricks.com/strategies-for-cache-busting-css/#article-header-id-2 
https://www.keycdn.com/support/what-is-cache-busting

probably it's not the case anymore

Reply:
I think the async chunks is the biggest deal hear, using a single unique base path feels like the most straightforward approach to me.

Reply:
one downside here is a patch release with modest changes(or none) will effectively invalidate all assets. 

i'm a fan of *.${file content hash}.min.js but in practice i'm sure it would be negligble.  several and conditions to cause a difference and there's the issue of connecting it all together in our build.

Reply:
> Also, it requires to tweak lazy loading for webpack async chunks.

good point.